src/meltano/core/state_store/google.py

```diff
@@ -2,9 +2,42 @@
 from __future__ import annotations
 
 import re
+from contextlib import contextmanager
+
+from cached_property import cached_property  # type: ignore
```

Comment:
Same as my other suggestion above.

```suggestion
import re
import sys
from contextlib import contextmanager

if sys.version_info >= (3, 8):
    from functools import cached_property
else:
    from cached_property import cached_property
```